cli/ceph/osd/crush.py

```diff
@@ -0,0 +1,162 @@
+from cli import Cli
+
+
+class Crush(Cli):
+    """This module provides CLI interface to manage the Crush service."""
+
+    def __init__(self, nodes, base_cmd):
+        super(Crush, self).__init__(nodes)
+
+        self.base_cmd = f"{base_cmd} crush"
+
+    def rule(self, *args):
+        """
+        To create rules
+        Args:
+            Supported args
+            rule_type (str): create-simple | create-replicated |create-erasure
+            rule_name (str): name of the rule
+            root (str): root of the CRUSH hierarchy
+            failure_domain_type (str): failure domain (host/rack)
+            device_class (str): storage device class (hdd/sdd)
+            replicated (bool): if the rule is replicated or not
+        """
+        cmd = f"{self.base_cmd} rule"
+        for arg in args:
+            cmd += f" {arg}"
```

Comment:
will unsupported parameters won't cause any issue ?

Reply:
Yes, but since there are many combinations, handling all in the code will be difficult. If the user specifies a wrong combination, the create rule command itself will fail.